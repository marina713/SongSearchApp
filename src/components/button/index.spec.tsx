import * as React from 'react';
import renderer from 'react-test-renderer';

import {Button, Props} from './';

jest.mock('./styles', () => ({
  Container: 'Container',
  ButtonText: 'ButtonText',
}));

const initialProps = {
  text: 'Hello Button',
  onPress: () => {},
};

const testComponent = (props?: Props) => {
  return renderer.create(<Button {...initialProps} {...props} />);
};

describe('<Button />', () => {
  it('renders correctly with default props', () => {
    const tree = testComponent().toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with custom text', () => {
    const tree = testComponent({
      ...initialProps,
      text: 'Some other text',
    }).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
