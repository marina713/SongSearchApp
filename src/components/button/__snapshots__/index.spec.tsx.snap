// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly with custom text 1`] = `
<Container
  onPress={[Function]}
>
  <ButtonText>
    Some other text
  </ButtonText>
</Container>
`;

exports[`<Button /> renders correctly with default props 1`] = `
<Container
  onPress={[Function]}
>
  <ButtonText>
    Hello Button
  </ButtonText>
</Container>
`;
